woocommerce/woocommerce-ios
# [Woo POS] Cache `ItemList` thumbnails. Retrieve from cache for `CartView`

Closes: #13915


## Description
This PR addresses the design feedback on https://github.com/woocommerce/woocommerce-ios/issues/13859 regarding cart images loading with a delay compared to item list images.

In order to resolve this, we rely on Kingfisher's built-in caching mechanism to store the Item List thumbnails temporarily, then retrieve them if needed for the Cart.

https://github.com/user-attachments/assets/c45417e3-566b-41c5-8269-959966408005

## Changes
- Added `CartItemRowViewModel`. This is a wrapper around the existing `CartItem`, and loads the appropriate image if can be located in cache.
- Updated the `ItemRowView` to render a cached image if this exists.
- Usage of KingFisher's cache via `ImageCache.default` as caching system, and `DefaultImageProcessor` to apply no special processing for cached images. The use of a custom processor also requires to update the caching setup, by using the default processor we can use the default image cache straight-away.

I have not added new unit tests. The only addition that we could test would be the `CartItemRowViewModel`, but the only methods that are exposed right now would be testing the internals of Kingfisher's caching.

## Testing information
- Load POS
- Tap on products with images that are loaded in the Item List
- Observe that the images for the products added to the cart show with no delay.
- For easier visuals, you can also add a hint like `.border(.red, width: 2.0)` to the Images loaded from cache, [here](https://github.com/woocommerce/woocommerce-ios/blob/e4b918b00c7728b951a31941a7489f63b9abb228/WooCommerce/Classes/POS/Presentation/ItemRowView.swift#L71)

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
